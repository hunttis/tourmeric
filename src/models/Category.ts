export interface Category {
  abbreviation: string;
  formats?: string;
  image?: string;
  imageSmall?: string | null;
  logo: string;
  name: string;
  type: string;
}

export interface UploadedFile {
  downloadURL: string;
  name: string;
}
