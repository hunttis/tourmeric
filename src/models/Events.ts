export interface TourmericEvent {
  category: string;
  createDate: string;
  date: string;
  endDate?: string;
  format: string | null;
  entryFee: string;
  eventType?: string;
  name: string;
  published?: boolean;
  time: string;
  rulesLevel: string;
  playerSlots: string;
  prizes: string;
  link?: string;
  notes: string;
  rules?: string;
}

export interface TourmericEventEntry {
  key: string;
  value: TourmericEvent;
}
