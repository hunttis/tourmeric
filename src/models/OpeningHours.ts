export interface OpeningHoursException {
  date: string;
  name: string;
  openingHours: string;
  status: string;
  open: boolean;
}
