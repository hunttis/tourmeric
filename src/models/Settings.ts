
export interface Settings {
  activeLocationImage?: string;
  activeLogo: string;
  browserTitle?: string;
  companyinfo?: string;
  dateFormat: string;
  features?: Features;
  introText?: string;
  location?: Location;
  openingHours?: OpeningHours;
  pageSubtitle: string;
  pageTitle: string;
  privacyPolicy?: string;
  showSponsors?: boolean;
  subtitleTextColor: string;
  theme: string;
  titleBarAngle: string;
  titleBarColor: string;
  titleBarColor2: string;
  titleBarPercentage: string;
  titleTextColor: string;
  footer?: {[key: string]: Footer};
  locationImage?: string;
}

interface Features {
  events: Feature;
  highlights: Feature;
  storeinfo: Feature;
}

interface Feature {
  active: boolean;
  introtext: boolean;
}

interface Location {
  address: string;
  directions: string;
  email: string;
  phone: string;
}


export interface OpeningHours {
  additionalinfo: string;
  monday: string;
  tuesday: string;
  wednesday: string;
  thursday: string;
  friday: string;
  saturday: string;
  sunday: string;
}

export interface Footer {
  link: string;
  image: string;
  text: string;
}
