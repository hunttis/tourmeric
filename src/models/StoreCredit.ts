export interface TourmericStoreCreditData {
  date: string;
  creditAddedByName: string;
  creditAddedBy: string;
  note: string;
  value: number;
  category: CreditCategories;
}

export interface StoreCreditCategory { [key: string]: string }

export const CATEGORY_RED: CreditCategories = 'red';
export const CATEGORY_YELLOW: CreditCategories = 'yellow';
export const CATEGORY_GREEN: CreditCategories = 'green';
export const CATEGORY_BLUE: CreditCategories = 'blue';
export const CATEGORY_WHITE: CreditCategories = '';

export type CreditCategories = 'yellow' | 'red' | 'green' | 'blue' | '';
