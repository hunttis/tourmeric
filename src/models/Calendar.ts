import { TourmericEvent } from './Events';

export interface Day {
  day?: string;
  dayOfWeek?: number;
  dayName?: string;
  dayString?: string;
  dayLink?: string;
  empty?: boolean;
  eventsForDay?: { key: string, value: TourmericEvent }[];
  ongoingEventsForDay?: { key: string, value: TourmericEvent }[];
}
