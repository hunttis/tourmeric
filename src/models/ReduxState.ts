import { Settings } from './Settings';
import { Category, UploadedFile } from './Category';
import { OpeningHoursException } from './OpeningHours';
import { TourmericEvent } from './Events';
import { TourmericStoreCreditData, StoreCreditCategory } from './StoreCredit';

export interface ReduxState {
  admin: any;
  firebase: FirebaseObject;
  locale: any;
  router: Router;
  editor: Editor;
}

interface Editor {
  returnLocation: string;
}

export interface FirebaseObject {
  data: FirebaseData;
  ordered: OrderedFirebaseData;
  auth: FirebaseAuth;
  profile: FirebaseProfile;
}

export interface Router {
  location: Location;
  action: string;
}

export interface Location {
  pathname: string;
  search: string;
  hash: string;
  key: string;
}

interface OrderedFirebaseData {
  events: [{ key: string, value: TourmericEvent }];
  eventsongoing: [{ key: string, value: TourmericEvent }];
  news: [{ key: string, value: SingleNewsItem }];
  users: [{ key: string, value: User }];
}

interface FirebaseData {
  settings: Settings;
  highlights: { [key: string]: HighLight };
  events: { [key: string]: TourmericEvent };
  eventsongoing: { [key: string]: TourmericEvent };
  categories: { [key: string]: Category };
  participations: { [key: string]: Participation };
  uploadedFiles: { [key: string]: UploadedFile };
  uploadedCategoryLogos: { [key: string]: UploadedFile };
  uploadedHighlightBanners: { [key: string]: UploadedFile };
  uploadedNewsImages: { [key: string]: UploadedFile };
  uploadedFooterItems: { [key: string]: UploadedFile };
  uploadedStoreinfoFiles: { [key: string]: UploadedFile };
  openinghoursexceptions: { [key: string]: OpeningHoursException };
  storecreditcategories: { [key: string]: StoreCreditCategory };
  users: { [key: string]: User };
  news: { [key: string]: SingleNewsItem };
  storecredit: { [key: string]: { [key: string]: TourmericStoreCreditData }};
}

export interface SingleNewsItem {
  createDate: string;
  date: string;
  linkName: string;
  link: string;
  image: string;
  active: boolean;
  name: string;
  text: string;
  summary: string;
}


export interface HighLight {
  active: boolean;
  image: string;
  createDate: string;
  date: string;
  name: string;
}

export interface Participation {
  [key: string]: ParticipationData;
}

export interface ParticipationData {
  date: string;
  firstName: string;
  lastName: string;
  userId: string;
  comment: string;
}

export interface FirebaseAuth {
  uid: string;
  displayName: string;
  photoURL: string;
  email: string | null;
  emailVerified: boolean;
  providerData: ProviderData;
  apiKey: string;
  isEmpty: boolean;
  isLoaded: boolean;
}

export interface ProviderData {
  displayName: string;
  email: string;
  photoURL: string;
  providerId: string;
  uid: string;
}

export interface FirebaseProfile {
  email: string;
  acceptedPrivacyPolicy?: string;
  acceptsNewsLetter?: boolean;
  avatarUrl: string;
  betaParticipation?: any;
  dciNumber: string;
  displayName: string;
  favoriteCategories?: string;
  firstName: string;
  lastName: string;
  landingPage?: PageOption;
  otherEmail?: string;
  providerData: ProviderData[];
  role: string;
  useOtherEmail?: boolean;
  isEmpty: boolean;
  isLoaded: boolean;
}

export type PageOption = 'today' | 'events' | 'storeinfo' | 'userinfo' | 'companyinfo' | 'admintools' | 'admintoolsevents' | 'adminsitesettings' | '';

export interface User {
  avatarUrl?: string;
  displayName?: string;
  providerData?: ProviderData;
  firstName: string;
  lastName: string;
  email: string;
  username: string;
  dciNumber?: string;
  role?: string;
  active: boolean;
}
