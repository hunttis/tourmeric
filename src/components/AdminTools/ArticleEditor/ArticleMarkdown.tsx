import React, { useState } from "react";
import firebase from "firebase/app";
import { FormattedMessage, IntlShape } from "react-intl";
import { MarkdownElement } from "../../../components/Common/MarkdownElement";
import { Article } from "../../../models/ReduxState";
import EditableTextarea from "../../../components/Common/EditableTextarea-container";
import EditableVerticalField from "../../../components/Common/EditableVerticalField-container";
import { UploadedFile } from "../../../models/Category";
import { ImagePickerWithCallback } from "../ImagePickerWithCallback";

interface Props {
  articleId: string;
  article: Article;
  uploadedArticleImages: [{ key: string; value: UploadedFile }];
}

export const ArticleMarkdown = ({
  articleId,
  article,
  uploadedArticleImages,
}: Props) => {
  const [imagePickerOpen, setImagePickerOpen] = useState<boolean>(false);

  if (!article) {
    return <div className="is-loading">Loading...</div>;
  }

  return (
    <div className="section">
      <div className="columns is-multiline">
        <div className="column is-4">
          <h1 className="title">
            <FormattedMessage id="editarticle" />
          </h1>
        </div>
        <div className="column is-2 has-text-right">
          <ImagePickerButton
            imagePickerOpen={imagePickerOpen}
            setImagePickerOpen={setImagePickerOpen}
          />
        </div>
        <div className="column is-6">
          <h1 className="title">
            <FormattedMessage id="preview" />
          </h1>
        </div>
        {!imagePickerOpen && (
          <div className="column is-6">
            <p className="content is-marginless">
              <FormattedMessage id="articletitle" />
            </p>
            <EditableVerticalField
              labelContent=""
              placeHolder="articletitle"
              defaultValue={article.title}
              path={`/articles/${articleId}`}
              targetName="title"
            />
            <p className="content is-marginless">
              <FormattedMessage id="articlecontent" />
            </p>
            <EditableTextarea
              labelContent=""
              placeHolder="contentplaceholder"
              defaultValue={article.content}
              path={`/articles/${articleId}`}
              rows={50}
              targetName="content"
            />
          </div>
        )}
        {imagePickerOpen && (
          <div className="column is-6">
            <ImagePickerWithCallback
              imageList={uploadedArticleImages}
              onClickCallback={(url: string) => {
                firebase.update(`/articles/${articleId}`, {
                  content: `![alt text](${url} "image hover text")\n\n${article.content}`,
                });
              }}
            />
          </div>
        )}
        <div className="column is-6" id="preview">
          <h1 className="title has-text-info has-text-centered">
            {article.title}
          </h1>
          <MarkdownElement markdown={article.content} />
        </div>
        <div className="column is-6 has-text-right">
          <a
            className="button is-small is-primary"
            target="_blank"
            rel="noopener noreferrer"
            href="https://www.markdownguide.org/cheat-sheet/"
          >
            <span className="icon has-text-success is-paddingless is-marginless">
              <i className="fas fa-question" />
            </span>
            <FormattedMessage id="help" /> -{" "}
            <FormattedMessage id="markdowncheatsheet" />
          </a>
        </div>
        <div className="column is-6">
          <FormattedMessage id="allbasicfeaturessupported" />
        </div>
      </div>
    </div>
  );
};

interface ImagePickerButtonProps {
  imagePickerOpen: boolean;
  setImagePickerOpen: (status: boolean) => void;
}

const ImagePickerButton = ({
  imagePickerOpen,
  setImagePickerOpen,
}: ImagePickerButtonProps) => (
  <>
    {!imagePickerOpen && (
      <button
        className="button has-icon-left"
        onClick={() => setImagePickerOpen(true)}
      >
        <span className="icon has-text-success is-paddingless is-marginless">
          <i className="fas fa-plus" />
        </span>
        <span className="icon is-paddingless is-marginless">
          <i className="fas fa-image" />
        </span>
        <span>
          <FormattedMessage id="addimage" />
        </span>
      </button>
    )}
    {imagePickerOpen && (
      <button
        className="button has-icon-left is-warning is-outlined"
        onClick={() => setImagePickerOpen(false)}
      >
        <span className="icon has-text-success is-paddingless is-marginless">
          <i className="fas fa-plus" />
        </span>
        <span className="icon is-paddingless is-marginless">
          <i className="fas fa-image" />
        </span>
        <span>
          <FormattedMessage id="closeimagepicker" />
        </span>
      </button>
    )}
  </>
);
