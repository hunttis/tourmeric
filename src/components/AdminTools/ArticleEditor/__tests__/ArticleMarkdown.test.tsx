import React from 'react';
import { shallow, render } from 'enzyme';
import { ArticleMarkdown } from '../ArticleMarkdown';
import { Article } from '~/models/ReduxState';

describe('ArticleMarkdown Tests', () => {

  const mockArticle: Article = {
    createDate: '01-01-2017',
    date: '01-01-2017',
    title: 'Test title',
    published: true,
  };

  it("has undefined markdown content and doesn't crash", () => {
    shallow(
      <ArticleMarkdown
        articleId="12345"
        article={mockArticle}
        uploadedArticleImages={[] as any}
      />,
    );
  });

  it('has empty markdown content', () => {
    const mockArticleWithContent = mockArticle;
    mockArticleWithContent.content = '';

    const result = render(
      <ArticleMarkdown
        articleId="12345"
        article={mockArticleWithContent}
        uploadedArticleImages={[] as any}
      />,
    );

    const preview = result.find('#preview');
    expect(preview).toMatchSnapshot();
  });

  it('has small article with title, subtitle and text', () => {
    const mockArticleWithContent = mockArticle;
    mockArticleWithContent.content = '# Title\n## Subtitle\nText\nMore text';

    const result = render(
      <ArticleMarkdown
        articleId="12345"
        article={mockArticleWithContent}
        uploadedArticleImages={[] as any}
      />,
    );

    const preview = result.find('#preview');
    expect(preview).toMatchSnapshot();
  });
});
