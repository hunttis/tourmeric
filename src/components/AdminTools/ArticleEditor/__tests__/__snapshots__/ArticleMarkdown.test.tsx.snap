// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleMarkdown Tests has empty markdown content 1`] = `
<div
  class="column is-6"
  id="preview"
>
  <h1
    class="title has-text-info has-text-centered"
  >
    Test title
  </h1>
  <div />
</div>
`;

exports[`ArticleMarkdown Tests has small article with title, subtitle and text 1`] = `
<div
  class="column is-6"
  id="preview"
>
  <h1
    class="title has-text-info has-text-centered"
  >
    Test title
  </h1>
  <div>
    <h1
      class="title"
      id="title"
    >
      Title
    </h1>
    

    <h2
      class="subtitle"
      id="subtitle"
    >
      Subtitle
    </h2>
    

    <p
      class="content has-text-justified"
    >
      Text
More text
    </p>
  </div>
</div>
`;
